src/main/kotlin/com/shortener/service/UrlServiceImpl.kt

```diff
@@ -0,0 +1,101 @@
+package com.shortener.service
+
+import com.shortener.domain.UrlEntry
+import com.shortener.domain.UrlEntryRepository
+import com.shortener.dto.UrlShortenRequest
+import com.shortener.dto.UrlShortenResponse
+import com.shortener.exception.InvalidInputUrl
+import com.shortener.utils.IdEncoder
+import org.apache.commons.validator.routines.UrlValidator
+import org.apache.commons.validator.routines.UrlValidator.ALLOW_LOCAL_URLS
+import org.springframework.stereotype.Service
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder
+import java.time.LocalDateTime
+import java.util.*
+
+private const val HTTP_SCHEMA = "http"
+private const val DEFAULT_HTTPS_SCHEMA = "https"
+private const val SCHEMA_SEPARATOR = "://"
+private const val URL_EXPIRES_DAYS_NO_USER = 7L
+
+@Service
+class UrlServiceImpl(private val urlEntryRepository: UrlEntryRepository, private val idEncoder: IdEncoder) :
+    UrlService {
+
+    override fun shortenUrl(request: UrlShortenRequest): UrlShortenResponse {
+        val fixedRequestUrl = addDefaultSchemaIfMissing(request.longUrl)
+
+        if (!isValidUrl(fixedRequestUrl)) {
+            throw InvalidInputUrl("The URL you provided is invalid")
+        }
+
+        val urlEntry = UrlEntry()
```

Comment:
the kotlin way of doing this initialization is
```kotlin
val urlEntry = UrlEntry().apply {
	longUrl = fixedRequestUrl
    createdAt = LocalDateTime.now()
}
```

Reply:
I am convinced this will work if those properties were not private